GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/DehydrateTests.cs

```diff
@@ -176,279 +176,264 @@ public void DehydrateShouldFailOnWrongDiskLayoutVersion()
         }
 
-        [TestCase]
+        [TestCase]
         public void FolderDehydrateFolderThatWasEnumerated()
         {
-            string pathToEnumerate = this.Enlistment.GetVirtualPathTo("GVFS");
-            this.fileSystem.EnumerateDirectory(pathToEnumerate);
-            string subFolderToEnumerate = Path.Combine(pathToEnumerate, "GVFS");
-            this.fileSystem.EnumerateDirectory(subFolderToEnumerate);
+            string folderToDehydrate = "GVFS";
+            string folderToEnumerateVirtualPath = this.Enlistment.GetVirtualPathTo(folderToDehydrate);
+            string folderToEnumerateBackingPath = this.Enlistment.GetBackingPathTo(folderToDehydrate);
+            this.fileSystem.EnumerateDirectory(folderToEnumerateVirtualPath);
+            string subFolderToEnumerate = Path.Combine(folderToDehydrate, "GVFS");
+            string subFolderToEnumerateVirtualPath = this.Enlistment.GetVirtualPathTo(subFolderToEnumerate);
+            string subFolderToEnumerateBackingPath = this.Enlistment.GetBackingPathTo(subFolderToEnumerate);
+            this.fileSystem.EnumerateDirectory(subFolderToEnumerateVirtualPath);
 
-            this.DehydrateShouldSucceed(new[] { $"GVFS {FolderDehydrateSuccessfulMessage}" }, confirm: true, noStatus: false, foldersToDehydrate: "GVFS");
+            this.DehydrateShouldSucceed(new[] { $"{folderToDehydrate} {FolderDehydrateSuccessfulMessage}" }, confirm: true, noStatus: false, foldersToDehydrate: folderToDehydrate);
             this.Enlistment.UnmountGVFS();
-            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
-            {
-                pathToEnumerate.ShouldNotExistOnDisk(this.fileSystem);
-            }
-            else
-            {
-                pathToEnumerate.ShouldBeADirectory(this.fileSystem);
-            }
-
-            subFolderToEnumerate.ShouldNotExistOnDisk(this.fileSystem);
+            this.CheckDehydratedFolderAfterUnmount(folderToEnumerateBackingPath);
+            subFolderToEnumerateBackingPath.ShouldNotExistOnDisk(this.fileSystem);
         }
 
-        [TestCase]
+        [TestCase]
         public void FolderDehydrateFolderWithFilesThatWerePlaceholders()
         {
-            string pathToReadFiles = this.Enlistment.GetVirtualPathTo("GVFS");
-            string fileToRead = Path.Combine(pathToReadFiles, "GVFS", "Program.cs");
-            using (File.OpenRead(fileToRead))
+            string folderToDehydrate = "GVFS";
+            string folderToReadFilesVirtualPath = this.Enlistment.GetVirtualPathTo(folderToDehydrate);
+            string folderToReadFilesBackingPath = this.Enlistment.GetBackingPathTo(folderToDehydrate);
+            string fileToRead = Path.Combine(folderToDehydrate, "GVFS", "Program.cs");
+            string fileToReadVirtualPath = this.Enlistment.GetVirtualPathTo(fileToRead);
+            string fileToReadBackingPath = this.Enlistment.GetBackingPathTo(fileToRead);

```

Comment:
Style suggestion:

I personally think I would find this code easier to follow if you moved all of the `BackingPath` variable down to after the call to `UnmountGVFS` and included a short comment about why the backing path is getting used.

(This comment applies to all tests that needed this change)

Maybe something like:

```
// Use the backing path because on some platforms
// the virtual path is no longer accessible after unmounting
```

Reply:
I'm thinking we should have a class like this to help with this:
```
private class TestPath
{
    public TestPath(GVFSFunctionalTestEnlistment enlistment, string basePath)
    {
        this.BasePath = basePath;
        this.VirtualPath = enlistment.GetVirtualPathTo(basePath);
        this.BackingPath = enlistment.GetBackingPathTo(basePath);
    }

    public string BasePath { get; }
    public string VirtualPath { get; }
    public string BackingPath { get; }
}
```

Reply:
Nice idea!  I think I've implemented this pretty much as you suggested -- see what you think.  Thanks very much for the suggestion!